reana_workflow_controller/rest.py

```diff
@@ -1269,6 +1270,94 @@ def open_interactive_session(workflow_id_or_name, interactive_session_type):  #
         return jsonify({"message": str(e)}), 500
 
 
+@restapi_blueprint.route('/workflows/<workflow_id_or_name>/close',
+                         methods=['POST'])
+def close_interactive_session(workflow_id_or_name):  # noqa
+    r"""Close an interactive workflow session.
+
+    ---
+    post:
+      summary: Close an interactive workflow session.
+      description: >-
+        This resource is expecting a workflow to close an interactive session
+        within its workspace.
+      operationId: close_interactive_session
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
+        - name: user
+          in: query
+          description: Required. UUID of workflow owner.
+          required: true
+          type: string
+        - name: workflow_id_or_name
+          in: path
+          description: Required. Workflow UUID or name.
+          required: true
+          type: string
+      responses:
+        200:
+          description: >-
+            Request succeeded. The interactive session has been closed.
+          schema:
+            type: object
+            properties:
+              message:
+                type: string
+          examples:
+            application/json:
+              {
+                "message": "The interactive session has been closed",
+              }
+        400:
+          description: >-
+            Request failed. The incoming data specification seems malformed.
+          examples:
+            application/json:
+              {
+                "message": "Malformed request."
+              }
+        404:
+          description: >-
+            Request failed. Either User or Workflow does not exist.
+          examples:
+            application/json:
+              {
+                "message": "Workflow 256b25f4-4cfb-4684-b7a8-73872ef455a1
+                            does not exist"
+              }
+        500:
+          description: >-
+            Request failed. Internal controller error.
+    """
+    try:
+        user_uuid = request.args["user"]
+        workflow = None
+        workflow = _get_workflow_with_uuid_or_name(workflow_id_or_name,
+                                                   user_uuid)
+        if workflow.interactive_session_name is None:
+            return jsonify(
+                {"message": "Workflow - {} has no open interactive session."
+                            .format(workflow_id_or_name)}), 404
+        kwrm = KubernetesWorkflowRunManager(workflow)
+        kwrm.stop_interactive_session()
+        workflow.interactive_session_name = None
+        workflow.interactive_session = None
+        current_db_sessions = Session.object_session(workflow)
```

Comment:
What do you think about moving this DB business inside the `stop_interactive_session()` implementation? We are already doing something similar for opening, we set the interactive session name and path in DB inside the method.